### ISSUE TITLE
typescript types?

### ISSUE THREAD

**Issue by khsuzan** [dotenvx/dotenvx#103]
I am using it in my typescript project and facing to configure it with typescript

**Comment by motdotla** [comment]
there is currently not typescript type files added. this is a pure js project. would you like to add them for us?

**Comment by feledori** [comment]
I could take that on... If you want it to stay js only. jsdoc would be the way. Probably would makes sense doing this after #102 .

**Comment by motdotla** [comment]
the project is still not 1.0 so a lot of the method signatures still could change. definitely wait for #102 and then wait maybe a little longer after that to let things settle. i'd like to do the same as how dotenv does it.

**Comment by motdotla** [comment]
@khsuzan can you explain why you need this? `dotenvx` is mainly a cli library.

if you are using the old school `require` methodology, also why do you need types? it just loads to process.env.

**Comment by feledori** [comment]
@motdotla I can understand where he's coming from... Had to inject some env vars base on an env set by the cli. Had to look at the source from lib to check the function name. Having autocomplete for these type of situation is a nice to have.

**Comment by motdotla** [comment]
ok, let's wait until after the `1.0.0` release.

**Comment by khsuzan** [comment]
That's great, let's wait for that.

**Comment by EKarpinsky** [comment]
> @khsuzan can you explain why you need this? `dotenvx` is mainly a cli library.
> 
> if you are using the old school `require` methodology, also why do you need types? it just loads to process.env.

Commenting to maybe push for this feature soon.

I use this in, as you call it, the `old school` method (not sure what's old school about it), by pulling in my `.env` programmatically in the code. I do this because I have a package which I am linking via `pnpm link --global` to use in my other packages as a util package. The package I am linking has some `.env` vars necessary to make it work, but the `.env` is located in the package's directory. To make it work from other directories, I would need to specify the path of the `.env` of the global package every time I run it as a `bin`. Calling `dotenvx.config()` in the code itself allows me to bypass this. Unfortunately, I then get a typescript warning in the import for `dotenvx` which I have to disable. 

Not a huge hassle but would be nice to have types for this for sure, seeing as calling `config()` in your code not deprecated and is certainly a valid use case.

**Comment by motdotla** [comment]
> To make it work from other directories, I would need to specify the path of the .env of the global package every time I run it as a bin. Calling dotenvx.config() in the code itself allows me to bypass this

makes total sense. thank you for sharing.

we're closing in on 1.0.0 release - probably 1 month away. then at that time, I think we will be ready to add types in.

**Comment by motdotla** [comment]
actually, i think we could tackle `lib/main.js` now if you guys want to. 

i would prefer jsdoc - that way it is tighly coupled with the code signature. when someone makes a change to the signature, they will see they should also adjust the jsdoc.

if someone tackles the PR for `lib/main.js` i'll get that merged in pretty much right away. let's keep it just to that though for now if you could. the underlying services and helpers could still end up changing a lot but i don't think `lib/main` signatures will.

**Comment by getlarge** [comment]
@motdotla I would be up for the task.
I would suggest 
- using a TS declaration file for ease of distribution
- import the type definitions  with JSDoc in the `lib/main.js`
- enable type checking in `lib/main.js` with `// @ts-check` 

Ok with that?
